addons/udes_stock/models/stock_location.py

```diff
@@ -106,9 +104,51 @@ def _domain_height_category(self):
 
                 If not set then the location will use the format
                 of the nearest ancestor that has a format specified.
-                """
+                """,
+    )
+    u_state = fields.Selection(
+        selection=[
+            ("empty", "Empty"),
+            ("blocked", "Blocked"),
+            ("has_stock", "Has Stock"),
+            ("archived", "Archived"),
+        ],
+        compute="_compute_state",
+        store=True,
+        string="State",
+        help="""Computed field describing the state of the stock location.""",
     )
 
+    @api.depends("quant_ids", "usage", "active")
+    def _compute_state(self):
+        """
+        Determine the state of the stock location - archived, empty or has_stock.
+        The computed value is only set properly once a location has been created or saved.
+        This is because for on the fly changes the location in self will only have access
+        to fields in the location form, which doesn't include quant_ids. This can lead to the
+        computed value temporarily displaying empty before being set to has_stock once the record
+        is saved.
+
+        TODO: Handle blocked locations
+        """
+        Quant = self.env["stock.quant"]
+
+        for location in self:
+            state = False
+
+            if location.id and location.usage == "internal":
+                if not location.active:
+                    state = "archived"
+                # TODO: Not implemented yet
```

Comment:
I thought we had a simple blocking mechanism, will take a look.

Reply:
Blocking is currently done via `udes_simple_location_blocking`, so we'd need to override this method in that module. Going to be a bit awkward though as we'd want to keep this order intact.

Reply:
Sorry, I didn't realise this was wanted for some reason i read this and was like that is a later todo. Added the last two commits for this. (Have force pushed another - the if/elif logic is in a different method that overrides it)